PowerEditor/src/Utf8_16.cpp

```diff
@@ -152,7 +152,7 @@ size_t Utf8_16_Read::convert(char* buf, size_t len)
         case uni16LE_NoBOM:
         case uni16BE:
         case uni16LE: {
-            size_t newSize = len + len / 2 + 1;
+            size_t newSize = len + len / 2 + 64;

```

Comment:
I meant:
Why 64 instead of 1 ?

Reply:
because, with one, some bug in `m_Iter16` writes beyond the buffer.
See https://github.com/notepad-plus-plus/notepad-plus-plus/issues/9456#issuecomment-1086681301

Reply:
@somanuell 
I've read the issue. 
My question was why 64 instead of 1 fixes the crash.

Reply:
No idea. I first tried with `size_t newSize = len * 2;` but in the end I found it a bit overkill...
The fundamental problem is that:

1. The file is binary
2. The fact that `IsTextUnicode` returns TRUE for the 131076 first bytes doesn't mean anything sensible
3. When `Utf8_16_Read::convert` is called for the 88037 remaining bytes, it triggers an overwrite in `Utf16_Iter m_Iter16;`

I just don't know if it's possible to code a conversion with random bytes. Maybe `Utf16_iter`  **should detect** that something is wrong with the so-called utf16 bytes, which are not.

I will maybe provide another PR. If I have time, I will investigate the bug in `Utf16_iter`...

In the meantime, I have a perfectly reproducible bug in the category "write beyond allocated heap buffer", that's pretty bad, security wise.

You should patch.

Reply:
"No idea" of the solution you provided cannot be taken as a valid argument.
And I cannot patch a magic number that you have tried and found because it will cause surely some regressions.

Regarding `BUILD.md` part, I will take care personally.

Reply:
The solution I came up with is to increase the size of the allocation, **since the current magic len + len / 2 + 1 is insufficient and crash!**

gosh...

bye, then.

Reply:
Please, don't get angry, I get why donho is asking about the allocation. Everytime I fixed something without fully understanding why it happened it ended up bitting me in the rear later. He wants to understand exactly why this is happening because using a number that works now may not be enough in other cases or even with other files.

Reply:
This is happening because there is a bug in Utf16_Iter that overwrites unallocated heap memory... Listen, I don't care now!
keep the magic len + len / 2 + 1 and let's just let the thing crash